backend/endpoints/rooms.py

```diff
@@ -41,19 +41,22 @@ async def create_room(
         HTTP_200_OK (room already exist): {room_id}
         HTTP_201_CREATED (new room created): {room_id}
         HTTP_424_FAILED_DEPENDENCY: room creation unsuccessful
+        HTTP_406_NOT_ACCEPTABLE: max number of dm users exceeded
     """
     room_data = request.dict()
-    room_member_ids = room_data["room_members"]
-    user_rooms = await get_rooms(member_id, org_id)
-    print("rooms", user_rooms)
+    room_members = set(room_data["room_members"])
+    if len(room_members) > 9:
+        return JSONResponse(
+            content={"message": "Cannot have more than 9 users in a DM"},
+            status_code=status.HTTP_406_NOT_ACCEPTABLE,
+        )
+    user_rooms = await get_rooms(org_id)  # add member_id after adding it to function
     if isinstance(user_rooms, list):
```

Comment:
the instance of user_rooms will always be a list.

Reply:
In some extreme cases, a dictionary or None is returned, which leads to code-breaking. Hence the need for extra validation.

Reply:
A case where a contributor decides to change the return value of the get_room function (e.g return None or 404 if the rooms are not found, or return the database response directly).
This will cause the loop and dictionary slicing methods used in the create_room function to throw an exception.

The idea of validating is to ensure that the function is independent. And not broken by unexpected responses from the called function. 

I am particularly conscious of this because of previous experiences during the internship, where the endpoint failed at very critical moments due to these scenarios.

Reply:
I understand... We definitely need testing now.

Reply:
I understand... We definitely need testing now.